### DISCUSSION TITLE
Best practice for generating normal map and roughness map mips?

### DISCUSSION THREAD

**Discussion by BStringhamVRSK** [google/filament#6260] [General]
Does Filament provide a best practice workflow for generating normal and roughness maps?
For a normal map with a lot of variation, naturally becomes "smoother" at each lower mip level due to averaging.  The information lost by that smoothing should ideally be captured by increasing the roughness of the corresponding roughness mip map.

Does an automated process already exist to do this with Filament tools?

**Comment by romainguy** [comment]
You can use the `roughness-prefilter` tool we provide for this  exact case.

**Reply by BStringhamVRSK** [reply]
Excellent!  I hadn't noticed that tool previously.
Thanks for the quick reply.

**Reply by BStringhamVRSK** [reply]
Once I have all of those individual mip files, I need to bundle them into a KTX result.  I'm currently using **mipgen** to both create the mips and bundle them into the KTX.  Can mipgen take prebuilt mips and process them into a KTX?  I've been looking at the command-line arguments, but I'm not seeing an obvious way to do that.
